core/src/driver/stablex_driver.rs

```diff
@@ -189,17 +216,49 @@ impl<'a> StableXDriver for StableXDriverImpl<'a> {
     }
 }
 
+fn burnt_fees(orders: &[Order], solution: &Solution) -> Result<U256> {
+    let orders: HashMap<u16, &Order> = orders.iter().map(|order| (order.id, order)).collect();
+    let mut fee_token_imbalance = BigInt::zero();
+    for executed_order in solution.executed_orders.iter() {
+        let id = executed_order.order_id;
+        let order = orders
+            .get(&id)
+            .ok_or_else(|| anyhow!("order {} not found", id))?;
+        if order.sell_token == 0 {
+            fee_token_imbalance += executed_order.sell_amount;
+        }
+        if order.buy_token == 0 {
+            fee_token_imbalance -= executed_order.buy_amount;
+        }
+    }
+    bigint_u256::bigint_to_u256(&fee_token_imbalance)
+        .ok_or_else(|| anyhow!("invalid burnt fee amount: {}", fee_token_imbalance))
+}
+
+fn gas_price_cap(burnt_fees: U256, num_trades: usize) -> U256 {
```

Comment:
Maybe we can add a unit test that documents how much GWEI we would be willing to pay e.g. for our 50k market order.